ocaml/xapi/xapi_vdi.ml

```diff
@@ -1007,6 +1007,10 @@ let list_changed_blocks ~__context ~vdi_from ~vdi_to =
   let task = Context.get_task_id __context in
   (* We have to pass the SR of vdi_to to the SMAPIv2 call *)
   let sr = Db.VDI.get_SR ~__context ~self:vdi_to in
+  (* assert VDIs on same SR *)
+  if sr <> (Db.VDI.get_SR ~__context ~self:vdi_from)
+  then raise (Storage_interface.Vdis_on_different_srs (Ref.string_of vdi_from, Ref.string_of vdi_to));
```

Comment:
I think Jon meant that we should throw this SMAPIv2 error from the SMAPIv2 layer, which is `storage_access.ml`.

Reply:
Then we should catch that in the xapi layer and raise a corresponding api error.

Reply:
exactly